### DISCUSSION TITLE
S2500 Inverter

### DISCUSSION THREAD

**Discussion by gjhill6100** [nathanmarlor/foxess_modbus#231] [General]
Hi,

Will this integration also work with a S2500 inverter ?

Looking to monitor it using the Com Port RS485 connections which I think are pins 3 & 4 with a RS485 to LAN converter.

Many thanks for any help.

**Comment by calum-mcfarlane** [comment]
I don't want to speak for the devs but I suspect you might be about to get asked to help them develop support ;O)

(eg #171)

The thing here is that different inverter models can use different registers for different things, and it's a matter of trial and error to work it out for each different one, since Fox don't make documentation readily available.

**Comment by canton7** [comment]
Yep, @calum-mcfarlane nailed it :)

I'd love to add S-series support, but I've never seen or used one, and I don't know what registers it exposes (this information isn't published by FoxESS: people figure it out by trial and error).

So by all means, please connect up an RS485 adapter and see if you can get it to respond using a tool such as modpoll. From there we can work together to see what registers if exposes, and if they're in any way similar to the H-series.

**Reply by gjhill6100** [reply]
I need to get the S2500 communicating in the first place. Not sure if the Connections 3 & 4 are correct or how to configure the Inverter to just send out RS485 on the relevant pins.

![S2500 Com Connector Pins](https://github.com/nathanmarlor/foxess_modbus/assets/120195446/1c491fa3-dfe5-401b-a870-8e36fd02a9bb)

**Reply by canton7** [reply]
That looks similar to the E/F series, see #220.

I know the manual has a small mention of connecting RS485 to a PC, but I'm not convinced. The METER 485 connection is for connecting to an electricity meter instead of a CT clamp, and on that bus the inverter will be the master. However, the H-series has a separate bus which exposes the registers we care about, and on that bus the inverter acts as a slave. Modbus only allows one master on a given connection.

**Reply by canton7** [reply]
I've just learnt that the T-series doesn't expose a modbus interface, but instead just transmits some statistics every few mintues. Maybe the S/E/F do the same?
